textile/fabric_pretreatment/doctype/pretreatment_order/pretreatment_order.py

```diff
@@ -22,7 +22,7 @@
 )
 
 force_fields = force_customer_fields + force_fabric_fields + force_process_component_fields
-
+ignore_permissions = frappe.has_permission("Pretreatment Order", "write")
```

Comment:
Very bad idea to keep a global variable here. Instead check permissions where ever necessary by `frappe.has_permission()` or `frappe.check_permission()`. And revert back to `ignore_permissions=True`